src/Common/src/Internal/Runtime/EETypeOptionalFieldsBuilder.cs

```diff
@@ -93,7 +93,15 @@ public override string ToString()
 
             for (int i = 0; i < (int)EETypeOptionalFieldsElement.Count; i++)
             {
-                sb.Append(_rgFields[i]._value.ToString());
+                if (_rgFields[i]._fieldPresent)
+                {
+                    sb.Append(_rgFields[i]._value.ToString());
```

Comment:
EETypeOptionalFieldsBuilder is not part of ILCompiler - it's shared with ProjectN. Do we want to move the FormattingHelpers into Common?